src/python/pants/build_graph/meta_rename.py

```diff
@@ -0,0 +1,69 @@
+# coding=utf-8
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from collections import defaultdict
+
+from pants.backend.jvm.targets.scala_library import ScalaLibrary
+from pants.base.specs import DescendantAddresses
+from pants.build_graph.address import Address
+from pants.task.task import Task
+
+class MetaRename(Task):
+  """Rename a target for its dependants
+
+  Provides a mechanism for renaming the target's name within its local BUILD file.
+  Also renames the target for its addresses wherever it's specified as a dependency.
+  """
+
+  @classmethod
+  def register_options(cls, register):
+    super(MetaRename, cls).register_options(register)
+
+    register('--from', type=str, advanced=True, default=None, help='The old dependency name to change')
+    register('--to', type=str, advanced=True, default=None, help='The new name for the dependency')
+
+  def __init__(self, *args, **kwargs):
+    super(MetaRename, self).__init__(*args, **kwargs)
+
+    self._from = self.get_options()['from']
+    self._to = self.get_options().to
+
+  def execute(self):
+    self.update_target_builds()
+
+  def update_target_builds(self):
+    from_address = Address.parse(self._from)
+    to_address = Address.parse(self._to)
+    
+    self.replace_in_file(from_address.spec_path + '/BUILD', from_address.target_name, to_address.target_name)
+
+    from_target = ScalaLibrary(name=from_address.target_name, address=from_address, build_graph=[], **{})
+    dependency_graph = self.dependency_graph()
+    dependant_addresses = dependency_graph[from_target]
+    
+    for address in dependant_addresses:
+      self.replace_in_file(address.spec_path + '/BUILD', from_address.target_name, to_address.target_name)
+
+  def dependency_graph(self, scope=''):
+    dependency_graph = defaultdict(set)
+
+    for address in self.context.build_graph.inject_specs_closure([DescendantAddresses(scope)]):
+      target = self.context.build_graph.get_target(address)
+
+      for dependency in target.dependencies:
+        dependency_graph[dependency].add(address)
+
+    return dependency_graph
+
+  def replace_in_file(self, _file, old_name, new_name):
```

Comment:
This is a start for sure, but still a bit far from what's expected. 
E.g.
```
./pants meta-rename --from=examples/src/java/org/pantsbuild/example/hello/greet --to=examples/src/java/org/pantsbuild/example/hello/greet:x
```

should replace `examples/src/java/org/pantsbuild/example/hello/greet(:greet)` to `examples/src/java/org/pantsbuild/example/hello/greet:x`, but the product contains unexpected result:
```
--- a/examples/tests/java/org/pantsbuild/example/hello/greet/BUILD
+++ b/examples/tests/java/org/pantsbuild/example/hello/greet/BUILD
@@ -1,11 +1,11 @@
 # Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-# Test the Hello World example's "greet" library
+# Test the Hello World example's "x" library
 
 junit_tests(
   dependencies=[
-    'examples/src/java/org/pantsbuild/example/hello/greet',
+    'examples/src/java/org/pantsbuild/example/hello/x',
     'examples/src/resources/org/pantsbuild/example/hello',
   ],
 )
```
To solve this problem, the task needs to have a better understanding of the BUILD, which I will cover in the discussion tomorrow with [buildozer](https://github.com/bazelbuild/buildtools/tree/master/buildozer)